### DISCUSSION TITLE
Is it expected that using JUnit5 Assumptions in mixed JUnit4/5 Test bundles cause builds to fail in Tycho 2.6, while running fine in Tycho 2.5?

### DISCUSSION THREAD

**Discussion by beyerjsb** [eclipse-tycho/tycho#616] [Q&A] [answered]
Currently trying to find out if this is a bug in Tycho 2.6+ or not.

Test bundle with mixed JUnit4/5 tests inside. 
One of the JUnit5 test classes uses org.junit.jupiter.api.Assumptions.assumeTrue(<expression>).
In Tycho 2.5 this builds fine, the test is marked as skipped, as expected.
In Tycho 2.6 this fails the build with:
`org.opentest4j.TestAbortedException: Assumption failed: assumption is not true`

Resolution was to use org.junit.Assume.assumeTrue(<expression>), thus making the test not pure JUnit5 anymore.

Difference between Tycho 2.5 and 2.6 is that in 2.6 it uses the new JUnit57vintage provider.

If I use the JUnit5 Assumption in a pure JUnit5 test bundle, the JUnit5 assumption works as expected again, and it uses the JUnit57 provider but not the JUnit57vintage provider.

Did I miss something that is necessary since Tycho 2.6 or is this a bug?

Thanks.

**Comment by beyerjsb** [comment, marked as answer]
After a little more investigation I'm fairly confident this is a bug in Tycho 2.6+. I'm going to convert this to an issue.

**Reply by mickaelistria** [reply]
Please submit a minimal project to reproduce this issue, ideally already integrated as an intergration test in the tycho-its modules.

**Comment by szarnekow** [comment]
For future reference: The ticket: https://github.com/eclipse/tycho/issues/630
